packages/forklift-console-plugin/src/modules/Plans/views/details/components/PlanPageHeadings.tsx

```diff
@@ -44,10 +58,25 @@ export const PlanPageHeadings: React.FC<{ name: string; namespace: string }> = (
   const buttonStartIcon = canReStart ? <ReStartIcon /> : <StartIcon />;
 
   const criticalCondition =
-    loaded &&
-    !error &&
+    planLoaded &&
+    !planError &&
     plan?.status?.conditions?.find((condition) => condition?.category === 'Critical');
 
+  /**
+   * Check if the preserve static IPs is enabled with pod networking mapping.
+   */
+  const preserveIpsWithPodMapCondition = (): boolean => {
+    if (!netMapsLoaded || netMapsError) return false;
+
+    const isPreserveStaticIPs = plan?.spec?.preserveStaticIPs;
+    const planNetMaps = netMaps
+      ? netMaps.find((net) => net?.metadata?.name === plan?.spec?.map?.network?.name)
+      : null;
```

Comment:
`(netMaps || []).find((net) => net?.metadata?.name === plan?.spec?.map?.network?.name)`

im guessing this will give you same result